test-common/src/main/java/org/apache/kafka/common/test/TestUtils.java

```diff
@@ -125,4 +148,131 @@ static void retryOnExceptionWithTimeout(final Runnable runnable) throws Interrup
             Thread.sleep(DEFAULT_POLL_INTERVAL_MS);
         }
     }
+
+    /**
+     * Wrap a single record log buffer.
+     */
+    public static MemoryRecords singletonRecords(byte[] value,
```

Comment:
If those helpers are used by `DeleteTopicTest` only, could you please move them into `DeleteTopicTest`?

Reply:
These helper methods come from `TestUtils.scala`. In the Scala version, they are widely used across many test classes. Considering that we will eventually drop Scala, I think we could retain these helper methods in `TestUtils.java`.